From: NeilBrown <neilb@suse.de>
Date: Mon, 2 Mar 2015 17:02:29 +1100
Subject: [PATCH 25/57] md/bitmap: fix incorrect DIV_ROUND_UP usage.
Git-commit: 935f3d4fc62c1f4d99cd13344762e766cd3bf115
Patch-mainline: v4.1
References: fate#316335

DIV_ROUTND_UP doesn't work on "long long", - and it should be
sector_t anyway.

Signed-off-by: NeilBrown <neilb@suse.de>
Acked-by: NeilBrown <neilb@suse.com>

---
 drivers/md/bitmap.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/md/bitmap.c b/drivers/md/bitmap.c
index ebba40228c7b..a8d9450d9b6c 100644
--- a/drivers/md/bitmap.c
+++ b/drivers/md/bitmap.c
@@ -565,11 +565,11 @@ static int bitmap_read_sb(struct bitmap *bitmap)
 re_read:
 	/* If cluster_slot is set, the cluster is setup */
 	if (bitmap->cluster_slot >= 0) {
-		long long bm_blocks;
+		sector_t bm_blocks;
 
 		bm_blocks = bitmap->mddev->resync_max_sectors / (bitmap->mddev->bitmap_info.chunksize >> 9);
 		bm_blocks = bm_blocks << 3;
-		bm_blocks = DIV_ROUND_UP(bm_blocks, 4096);
+		bm_blocks = DIV_ROUND_UP_SECTOR_T(bm_blocks, 4096);
 		bitmap->mddev->bitmap_info.offset += bitmap->cluster_slot * (bm_blocks << 3);
 		pr_info("%s:%d bm slot: %d offset: %llu\n", __func__, __LINE__,
 			bitmap->cluster_slot, (unsigned long long)bitmap->mddev->bitmap_info.offset);
-- 
2.5.0

